crates/primitives/transactions/src/compute_hash.rs

```diff
@@ -58,60 +58,53 @@ impl Transaction {
-                    Pedersen::hash(&tx_hash, &signature_hash)
-                } else {
-                    let elements: Vec<Felt> = std::iter::once(tx_hash).chain(tx.signature().iter().copied()).collect();
-                    Poseidon::hash_array(&elements)
-                }
-            }
-            Transaction::Declare(tx) => {
-                if include_signature {
-                    if starknet_version < StarknetVersion::V0_13_2 {
-                        let signature_hash = tx.compute_hash_signature::<Pedersen>();
-                        Pedersen::hash(&tx_hash, &signature_hash)
-                    } else {
-                        let elements: Vec<Felt> =
-                            std::iter::once(tx_hash).chain(tx.signature().iter().copied()).collect();
-                        Poseidon::hash_array(&elements)
-                    }
-                } else {
-                    let signature_hash = Pedersen::hash_array(&[]);
-                    Pedersen::hash(&tx_hash, &signature_hash)
-                }
-            }
-            Transaction::DeployAccount(tx) => {
-                if include_signature {
-                    if starknet_version < StarknetVersion::V0_13_2 {
-                        let signature_hash = tx.compute_hash_signature::<Pedersen>();
-                        Pedersen::hash(&tx_hash, &signature_hash)
-                    } else {
-                        let elements: Vec<Felt> =
-                            std::iter::once(tx_hash).chain(tx.signature().iter().copied()).collect();
-                        Poseidon::hash_array(&elements)
-                    }
-                } else {
-                    let signature_hash = Pedersen::hash_array(&[]);
-                    Pedersen::hash(&tx_hash, &signature_hash)
-                }
-            }
-            _ => {
-                if starknet_version < StarknetVersion::V0_13_2 {
-                    let signature_hash = Pedersen::hash_array(&[]);
-                    Pedersen::hash(&tx_hash, &signature_hash)
-                } else {
-                    Poseidon::hash_array(&[tx_hash, Felt::ZERO])
-                }
-            }
+        if starknet_version < StarknetVersion::V0_11_1 {
+            self.compute_hash_with_signature_pre_v0_11_1(tx_hash)
+        } else if starknet_version < StarknetVersion::V0_13_2 {
+            self.compute_hash_with_signature_pre_v0_13_2(tx_hash)
+        } else {
+            self.compute_hash_with_signature_latest(tx_hash)
+        }
+    }
+
+    fn compute_hash_with_signature_pre_v0_11_1(&self, tx_hash: Felt) -> Felt {
+        let signature_hash = match self {
+            Transaction::Invoke(tx) => tx.compute_hash_signature::<Pedersen>(),
+            Transaction::Declare(_)
+            | Transaction::DeployAccount(_)
+            | Transaction::Deploy(_)
+            | Transaction::L1Handler(_) => Pedersen::hash_array(&[]),
```

Comment:
create a constant with this value `Pedersen::hash_array(&[])`

Reply:
Resolved.